src/shared_gui_components/EditView.cpp

```diff
@@ -400,4 +402,80 @@ void InputCheckBox::setIncomplete(bool incomplete) {
   }
 }
 
+PathInputView::PathInputView(const std::string& extension, bool isRead)
+  : InputView(), lineEdit(new QLineEdit()), nameLabel(new QLabel()), m_isRead(isRead) {
+  auto* vLayout = new QVBoxLayout();
+  vLayout->setContentsMargins(0, 0, 0, 0);
+  vLayout->setSpacing(5);
+  setLayout(vLayout);
+
+  nameLabel->setOpenExternalLinks(true);
+  nameLabel->setWordWrap(true);
+  vLayout->addWidget(nameLabel);
+
+  auto* hLayout = new QHBoxLayout();
+  hLayout->addWidget(lineEdit);
+
+  selectPathButton = new QPushButton("...");
+  selectPathButton->setFlat(true);
+  selectPathButton->setFixedSize(30, 20);
+  selectPathButton->setObjectName("StandardGrayButton");
+
+  connect(selectPathButton, &QPushButton::clicked, this, &PathInputView::onSelectPathButtonClicked);
+  hLayout->addWidget(selectPathButton);
+
+  vLayout->addLayout(hLayout);
+
+  // TODO: sanitize the input?
+  m_extension = toQString(extension);
```

Comment:
TODO here. Basically I think OS SDK should store it as a readable format, and we can reconstruct a proper Qt files Filter from it.

Reply:
I think the important thing is to allow the user to ignore the file filter if they want to, so maybe always append "Any files (*)"?  The filters argument is a QStringList right?

Reply:
it's ";;" separated string.

Effectively, right now, until the SDK gets updated, the extension is always "All Files(*)"